### ISSUE TITLE
When should apps redownload external resources

### ISSUE THREAD

**Issue by 419monk** [Podcastindex-org/podcast-namespace#126]
Starting a conversation about when podcatchers, aggregators etc should redownload external resources.
Extranal resources in pc20-namespace include at least:
- transcripts
- chapters

As far as I can tell the current documentation does not take stand on when or if at all clients should check if the resources are updated. Is the general idea that the external resource's url should change whenever the content changes? Or is there some other way to tell the clients about the updates? Or are the external resources assumed to be redownloaded every time feed is parsed again?

Especially with the idea of community based chapters it is to be expected that the chapters-resource will change over time.

I think it would be a very good idea to think this through and document the expected and desired behavior of the client apps and aggregators.

**Comment by dellagustin** [comment]
from an app developer perspective, I would not download this resources when parsing the feed, but only when downloading the episode audio file.

Even so, if the user is connected I would check if they are updated (or simply download them again) when the user play the audio file, if network is available.
These are typically lightweight content, so I don't see a good reason not to get them again on demand when needed.

**Comment by npruzaniec** [comment]
I think there's a little more to this though as many podcasters will likely rely on crowd-sourced chapters and transcripts, or at least have a delay to adding these. If these are only checked/downloaded with the audio file, they would not get updated if changes are made. I know this is the same with the actual audio file, but the other assets around the file seem more likely to change.

Maybe not in the scope of the Podcast Index Org to declare best practices on for app developers, but should be considered I think. This is why I suggest adding a query timestamp to these asset URLs to notify readers/aggregators when a file has changed that it may have already grabbed. Would require these apps to re-download the feed file when the episode is played for that to be useful though.

**Comment by kilobit** [comment]
Agreeing with @dellagustin - HTTP already has effective caching / update mechanisms in place.

**Comment by thomasrynne** [comment]
I'm mostly commenting here to try to bump this old issue. From an app developer point of view it's still not clear to me when chapters and transcripts should be polled. I think there is currently an ambiguous interaction between the apps and the hosts. My current approach is to re-read when the url changes but I believe some hosts will change the content without changing the url.
It would be good to resolve this soon while the number of apps & hosts supporting these features is relatively low
This will also be an issue if more external resources are added by other specs (for example cross app comments #505)

**Comment by francosolerio** [comment]
I agree 100% with @thomasrynne. With the current specs Castamatic has to re-download chapters and transcripts every time the user resumes playing the episode. This is wasting bandwidth for both the user and the podcaster. Someone suggests it's also leaking important listener metadata (mobile IP address) more than necessary.

Making changing the URL mandatory for each revision of the external files might be a solution. Adding a version/UUID parameter to the tag might be better alternative.

**Comment by adamc199** [comment]
This is me. My chapters and transcripts are susually complted 1-8 hours after the episode is released.
I was under the impression podping could be a solution with the <reason> code?

**Comment by francosolerio** [comment]
@adamc199 For Castamatic it will be when podping will be usable from a serverless stack. Until then changing something (the chapters url or a chapter file UUID) in the feed RSS would be the most easy way to proceed for both the apps and podcasters who delegate chapter/transcript compilation.

**Comment by Marzal** [comment]
Hi, as per https://github.com/Podcastindex-org/podcast-namespace/discussions/749 and https://github.com/Podcastindex-org/podcast-namespace/discussions/695 we are closing old issues.

Please open a new [Discussions](https://github.com/Podcastindex-org/podcast-namespace/discussions) if you think that the matter it's worth pursuing.
